src/box/xrow.c

```diff
@@ -429,6 +430,7 @@ iproto_reply_ok(struct obuf *out, uint64_t sync, uint32_t schema_version)
 	}
 	iproto_header_encode(buf, IPROTO_OK, sync, schema_version, 1);
 	buf[IPROTO_HEADER_LEN] = 0x80; /* empty MessagePack Map */
+	flightrec_write_response(buf, IPROTO_HEADER_LEN + 1);
```

Comment:
Can you move it to `tx_end_msg`?

Having this function call in so many places is bad, because one can easily forget to call it when a new method is added.

Reply:
I guess yes, I thought about it: we can cast `msg->wpos`  to `iproto_header_bin` and fetch `v_len` - that would be request size.

Reply:
I've added savepoint to `iproto_msg` which is set during `tx_accept_msg`. Then in `tx_end_msg` we use this savepoint to restore the whole response.

Also I've changed the order of flight recorder & say logger initialization - I've swapped them since now flight recorder sets log callback - it should be done after flight recorder is inited.

Reply:
Note that i've also fixed `flightrec_metrics_interval` - it should be double since interval can be less than 1 second.

Reply:
As @alyapunov pointed out - solution to assign savepoint in `tx_accept_msg` is wrong since TX may yield and several responses would point to the same buffer position. So instead I patched `tx_end_msg` so that now it accepts savepoint. That savepoint is created after TX processed request and we are about to dump response. IDK whether it looks better than original patch or not. Can revert it tho and dump request to flight recorder right in `*_reply_*` functions family (as it was done in the first patch).

Reply:
IMO passing `svp` to `tx_end_msg` is better than inserting flightrec stubs into all `iproto_reply_` functions, because the latter is easier to overlook when a new `iproto_reply_` function is added.